index.rb

```diff
@@ -1,160 +1,250 @@
- #Search "ruby understanding self and main"
- 
- 
- 
- module Enumerable
-  def my_each
-  
-    for i in self 
-       yield(i)
-    end
-
+module Enumerable
+  def my_each(&block)
+    each(&block)
   end
 
   def my_each_with_index
-  
-    for i in self
-      yield(i, self.find_index(i))
-    end
+    return to_enum(:my_each_with_index) unless block_given?
 
+    each do |i|
+      yield(i, find_index(i))
+    end
   end
```

Comment:
- [ ]  There is a failing test on your `my_each_with_index` It doesn't return the correct index and value when tested with the original `each_with_index` 
P.S You are not meant to use the original enumerable methods in your customer method. The idea was for you to create your own method that behaves like the original one and not to use the original method.